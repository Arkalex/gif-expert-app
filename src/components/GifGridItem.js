import React from 'react'

export const GifGridItem = ( { id, title, url } ) => {
    return (
        <div className="card animate_animated animate_fadeIn">
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}
