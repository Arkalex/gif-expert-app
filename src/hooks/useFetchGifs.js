import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';


export const useFetchGifs = ( category ) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        getGifs()
            .then(imgs => {
                setState({
                    data: imgs,
                    loading: false
                })
            })
    }, [])
    return state;
}