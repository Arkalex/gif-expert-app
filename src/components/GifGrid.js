import React, { useState, useEffect } from 'react'
import { GifGridItem } from './GifGridItem';
import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect( ()=> {
        getGifs(category)
            .then(imgs => setImages(imgs))
    }, [category])

    

    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
            {
                images.map((img) => {
                    return <GifGridItem 
                                key={ img.id }
                                { ...img } 
                            />
                })
            }
            </div>
        </>
    )
}
