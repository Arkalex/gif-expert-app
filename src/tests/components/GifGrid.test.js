import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el <GifGrid />', () => {

    const category = 'Goku';
    test('should show correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />);

        expect( wrapper ).toMatchSnapshot();
        
    })

    test('should show items when img loaded with useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https:localhost',
            title: 'random'
        }]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow(<GifGrid category={category} />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
    })
    
    
})