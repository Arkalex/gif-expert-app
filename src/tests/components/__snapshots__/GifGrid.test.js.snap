// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> should show correctly 1`] = `
<Fragment>
  <h3
    className="animate_animated animate_fadeIn"
  >
    Goku
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> should show items when img loaded with useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate_animated animate_fadeIn"
  >
    Goku
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="random"
      url="https:localhost"
    />
  </div>
</Fragment>
`;
